ext/mbstring/libmbfl/filters/mbfilter_gb18030.c

```diff
@@ -400,7 +416,6 @@ static size_t mb_gb18030_to_wchar(unsigned char **in, size_t *in_len, uint32_t *
 			*out++ = c;
 		} else if (c > 0x80 && c < 0xFF && p < e) {
 			unsigned char c2 = *p++;
```

Comment:
Is ``0x80`` an invalid codepoint? It might make more sense to check for the invalid conditions just before so that this case doesn't need to be nested (which IMHO makes it easier to follow the control flow by guarding errors early and just having some continue statements)

Reply:
`c` is not a "codepoint" here but a "byte", and yes, `0x80` is not a valid leading byte for a GB18030 character.

It's a good idea to check for invalid leading bytes first, I can do that.